src/Symfony/Component/Yaml/Inline.php

```diff
@@ -270,97 +264,115 @@ private static function dumpArray($value, $flags)
         return sprintf('{ %s }', implode(', ', $output));
     }
 
+    private static function parseValue(StringReader $reader, $flags = 0, $delimiters = null, $evaluate = true, $references = array())
+    {
+        $reader->consumeWhiteSpace();
+        if ($reader->readChar('[')) {
+            return self::parseSequence($reader, $flags, $references);
+        }
+        if ($reader->readChar('{')) {
+            return self::parseMapping($reader, $flags, $references);
+        }
+
+        $scalar = self::parseScalar($reader, $flags, $delimiters, $evaluate, $references);
+
+        return $scalar;
+    }
+
     /**
      * Parses a scalar to a YAML string.
      *
-     * @param string $scalar
-     * @param int    $flags
-     * @param string $delimiters
-     * @param array  $stringDelimiters
-     * @param int    &$i
-     * @param bool   $evaluate
-     * @param array  $references
+     * @param StringReader $reader
+     * @param int          $flags
+     * @param string       $delimiters
+     * @param bool         $evaluate
+     * @param array        $references
      *
      * @return string A YAML string
      *
      * @throws ParseException When malformed inline YAML string is parsed
      *
      * @internal
      */
-    public static function parseScalar($scalar, $flags = 0, $delimiters = null, $stringDelimiters = array('"', "'"), &$i = 0, $evaluate = true, $references = array())
+    public static function parseScalar(StringReader $reader, $flags = 0, $delimiters = null, $evaluate = true, $references = array())
     {
-        if (in_array($scalar[$i], $stringDelimiters)) {
-            // quoted scalar
-            $output = self::parseQuotedScalar($scalar, $i);
-
-            if (null !== $delimiters) {
-                $tmp = ltrim(substr($scalar, $i), ' ');
-                if (!in_array($tmp[0], $delimiters)) {
-                    throw new ParseException(sprintf('Unexpected characters (%s).', substr($scalar, $i)));
+        $unescaper = new Unescaper();
+        if ($reader->readChar('"')) {
+            return $unescaper->unescapeDoubleQuotedString($reader);
+        } elseif ($reader->readChar('\'')) {
```

Comment:
To follow Symfony CS IIRC `elseif` is useless if previous `if` have a `return` inside.